src/Storages/Streaming/StorageStream.cpp

```diff
@@ -991,9 +997,23 @@ StorageStream::ShardsToRead StorageStream::getRequiredShardsToRead(ContextPtr co
     {
         assert(query_info.seek_to_info);
         const auto & settings_ref = context_->getSettingsRef();
-        bool require_back_fill_from_historical = !isInmemory()
-            && (Streaming::isChangelogKeyedStorage(dataStreamSemantic()) || Streaming::isVersionedKeyedStorage(dataStreamSemantic())
-                || (query_info.seek_to_info->getSeekTo() == "earliest" && settings_ref.enable_backfill_from_historical_store.value));
+
+        /// We allow backfill from historical store the following scenarios:
+        /// 1) For non-inmemory keyed storage stream, we always back fill from historical store (e.g. VersionedKV, ChangelogKV)
+        /// 2) Do time travel with settings `enable_backfill_from_historical_store = true`
+        bool require_back_fill_from_historical = false;
+        if (!isInmemory() && Streaming::isKeyedStorage(dataStreamSemantic()))
+            require_back_fill_from_historical = true;
+        else if (!query_info.seek_to_info->getSeekTo().empty() && settings_ref.enable_backfill_from_historical_store.value)
+        {
+            if (!query_info.seek_to_info->isTimeBased() && query_info.seek_to_info->getSeekTo() != "earliest")
+                throw Exception(
```

Comment:
As mentioned, to provide best user experience, when seek to with sn, disable backfill from historical store automatically.